syncstorage/storage/__init__.py

```diff
@@ -307,6 +313,63 @@ def delete_items(self, userid, collection, items):
             CollectionNotFoundError: the user has no such collection.
         """
 
+    @abc.abstractmethod
+    def create_transaction(self, userid, collection):
+        """Creates a transaction for multi-POST batch uploads.
+
+        Args:
+            userid: integer identifying the user in the storage.
+            collection: name of the collection.
+
+        Returns:
+            The new transaction's ID as a BigInteger timestamp.
+        """
+
+    @abc.abstractmethod
+    def append_items_to_transaction(self, batchid, userid, collection, items):
+        """Creates or updates multiple items from a multi-POST batch to a
+        transaction.
+
+        Args:
+            batchid: big integer batch identifier for this transaction
+            userid: integer identifying the user in the storage.
+            collection: name of the collection.
+            items: a list of dicts giving data for each item.
+
+        Returns:
+            The new last-modified timestamp for the transaction.
+
+        Raises:
+            ConflictError: the operation conflicted with a concurrent write.
+        """
+
+    @abc.abstractmethod
+    def commit_transaction(self, batchid, userid, collection):
```

Comment:
For consistency with the other methods on this class, perhaps the order of arguments here should be `userid`, `collection`, `batchid`.